sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/AggregationQuerySuite.scala

```diff
@@ -1018,6 +1018,24 @@ abstract class AggregationQuerySuite extends QueryTest with SQLTestUtils with Te
     val agg2 = agg1.groupBy($"text").agg(sum($"avg_res"))
     checkAnswer(agg2, Row("a", BigDecimal("11.9999999994857142860000")))
   }
+
+  test("SPARK-29140: HashAggregateExec aggregating binary type doesn't break codegen compilation") {
+    val withDistinct = countDistinct($"c1")
+
+    val schema = new StructType().add("c1", BinaryType, nullable = true)
+    val schemaWithId = StructType(StructField("id", IntegerType, nullable = false) +: schema.fields)
```

Comment:
nit: curious why you don't just have schema?

Reply:
Ah that's missed. I copied the code from ObjectHashAggregateSuite and tried to minimize the code to reproduce and clean up, but missed this. Thanks for pointing out!